extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/MDCScope.java

```diff
@@ -33,6 +35,10 @@ public void close() {
         MDC.remove(TRACE_ID);
         MDC.remove(SPAN_ID);
         MDC.remove(SAMPLED);
+
+        if (toRestore != null && toRestore.span().context() instanceof JaegerSpanContext) {
```

Comment:
This certainly fixes the problem, but I am wondering whether it isn't better to just restore the old values of the keys that were removed. For instance they could be explicitly set by a user or other integration.

Reply:
Hmm, yea, we could do that...

Reply:
I've followed the `ThreadLocalScopeManager` / `ThreadLocalScope` example :-)

But I agree it's cleaner to just restore the MDC map content. I have also found that the test can be improved a bit. I will submit a subsequent PR.